frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/resize-rectangle-interaction.ts

```diff
@@ -66,47 +75,54 @@ export class ResizeRectangleInteraction extends PointerInteraction {
             return false;
         }
         const mouseCoordinate = event.coordinate;
-        const newXScale =
-            (mouseCoordinate[0]! - this.currentResizeValues.originCorner[0]!) /
-            (this.currentResizeValues.draggedCorner[0]! -
-                this.currentResizeValues.originCorner[0]!);
-        const newYScale =
-            (this.currentResizeValues.originCorner[1]! - mouseCoordinate[1]!) /
-            (this.currentResizeValues.originCorner[1]! -
-                this.currentResizeValues.draggedCorner[1]!);
-        this.currentResizeValues.feature
+        const { draggedCorner, originCorner, currentScale, feature } =
+            this.currentResizeValues;
+        const newXScale = this.calculateNewScale(
+            draggedCorner[0]!,
+            originCorner[0]!,
+            mouseCoordinate[0]!
+        );
+        const newYScale = this.calculateNewScale(
+            draggedCorner[1]!,
+            originCorner[1]!,
+            mouseCoordinate[1]!
+        );
+        feature
             .getGeometry()!
             .scale(
-                newXScale / this.currentResizeValues.currentScale!.x,
-                newYScale / this.currentResizeValues.currentScale!.y,
-                this.currentResizeValues.originCorner
+                newXScale / currentScale.x,
+                newYScale / currentScale.y,
+                originCorner
             );
         this.currentResizeValues.currentScale = { x: newXScale, y: newYScale };
         return true;
     }
 
+    private calculateNewScale(
+        draggedCorner: number,
+        originCorner: number,
+        mouseCoordinate: number
+    ) {
+        const oldXLength = draggedCorner - originCorner;
+        const newXLength = mouseCoordinate - originCorner;
```

Comment:
Please remove the X from those variable names, as it is also used for X coordinates.

Reply:
Fixed